h/api/storage.py

```diff
@@ -49,6 +49,15 @@ def fetch_annotation(session, id_):
         return None
 
 
+def fetch_ordered_annotations(session, ids):
+    if not ids:
+        return []
+
+    anns = session.query(models.Annotation).filter(models.Annotation.id.in_(ids)).all()
+    anns = sorted(anns, key=lambda a: ids.index(a.id))
```

Comment:
Yeah you're right. Especially since it has quite a dramatic impact even with low numbers like 200.